### DISCUSSION TITLE
Astyle cannot use

### DISCUSSION THREAD

**Discussion by mendax1234** [nvimtools/none-ls.nvim#113] [Q&A] [answered]
Hi, I try to use Astyle to format my C code.
```
        null_ls.builtins.formatting.astyle,
```
But seems that the none-ls says astyle is non-executable. But I can't find where to install astyle in Mason. But I do have found the in the `BUILTin.md` in none-ls, astyle is supported.
https://github.com/nvimtools/none-ls.nvim/discussions?discussions_q=Astyle+cannot+use

**Comment by mochaaP** [comment, marked as answer]
You need to install astyle to your system manually (either [from your distribution](https://repology.org/project/astyle/versions), or [from source](https://sourceforge.net/projects/astyle/files/latest/download)).

**Reply by mendax1234** [reply]
@mochaaP Thanks for your reply! Besides, may I ask by using none-ls, I can't autoformat my code while I writing my code. Like set the tabsize whatever. This plugin can only format my code when I use a command to trigger that format process right?

Edited:
I have create a new discussion about this problem.
